File: common/src/test/java/guepardoapps/lucahome/common/helper/UtilsUnitTest.java
package guepardoapps.lucahome.common.helper;

import org.junit.Test;

import guepardoapps.lucahome.common.utils.ByteStringHelper;

import static org.junit.Assert.assertEquals;

public class UtilsUnitTest {
    @Test
    public void byteStringConversion_isCorrect() throws Exception {
        String expectedValue = "HelloWorld!";
        char[] valueToConvert = {
                4968, 6969, 7452,
                7452, 7659, 6003,
                7659, 7866, 7452,
                6900, 2277
        };

        String actualValue = ByteStringHelper.Companion.readStringFromCharArray(valueToConvert);

        assertEquals(expectedValue, actualValue);
    }
}

File: common/src/test/java/guepardoapps/lucahome/common/helper/UtilsUnitTest.kt
package guepardoapps.lucahome.common.helper

import guepardoapps.lucahome.common.utils.ByteStringHelper
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it

import org.junit.Assert.assertEquals
import org.junit.platform.runner.JUnitPlatform
import org.junit.runner.RunWith

@RunWith(JUnitPlatform::class)
class UtilsUnitTest : Spek({

    describe("Unit tests for ByteStringHelper") {

        beforeEachTest { }

        afterEachTest { }

        it("conversion should be valid") {
            // Arrange
            val expectedValue = "HelloWorld!"
            val valueToConvert = charArrayOf(
                    4968.toChar(),
                    6969.toChar(),
                    7452.toChar(),
                    7452.toChar(),
                    7659.toChar(),
                    6003.toChar(),
                    7659.toChar(),
                    7866.toChar(),
                    7452.toChar(),
                    6900.toChar(),
                    2277.toChar()
            )

            // Act
            val actualValue = ByteStringHelper.readStringFromCharArray(valueToConvert)

            // Assert
            assertEquals(expectedValue, actualValue)
        }
    }
})